test/data/s3/test_s3.py

```diff
@@ -805,99 +808,113 @@ def test_put_exceptions(inject_failure_rate):
             s3.put_many([("foo", "bar")])
 
 
+@pytest.fixture
+def s3_upload_args():
+    return {
+        'ServerSideEncryption': 'AES256',
+        'ACL': 'private', 
+        'StorageClass': 'STANDARD_IA'
+    }
+
 @pytest.mark.parametrize("inject_failure_rate", [0, 10, 50, 90])
 @pytest.mark.parametrize(
     argnames=["s3root", "objs", "expected"], **s3_data.pytest_put_strings_case()
 )
-def test_put_many(inject_failure_rate, s3root, objs, expected):
-    with S3(s3root=s3root, inject_failure_rate=inject_failure_rate) as s3:
-        s3urls = s3.put_many(objs)
-        assert list(dict(s3urls)) == list(dict(objs))
-        # results must be in the same order as the keys requested
-        for i in range(len(s3urls)):
-            assert objs[i][0] == s3urls[i][0]
-    with S3(inject_failure_rate=inject_failure_rate) as s3:
-        s3objs = s3.get_many(dict(s3urls).values())
-        assert_results(s3objs, expected)
-    with S3(s3root=s3root, inject_failure_rate=inject_failure_rate) as s3:
-        s3objs = s3.get_many(list(dict(objs)))
-        assert {s3obj.key for s3obj in s3objs} == {key for key, _ in objs}
-
-    # upload shuffled objs with overwrite disabled
-    shuffled_objs = deranged_shuffle(objs)
-    with S3(s3root=s3root, inject_failure_rate=inject_failure_rate) as s3:
-        overwrite_disabled_s3urls = s3.put_many(shuffled_objs, overwrite=False)
-        assert len(overwrite_disabled_s3urls) == 0
-    with S3(inject_failure_rate=inject_failure_rate) as s3:
-        s3objs = s3.get_many(dict(s3urls).values())
-        assert_results(s3objs, expected)
+def test_put_many(inject_failure_rate, s3root, objs, expected, s3_upload_args):
+    upload_args = [s3_upload_args, None]
+    for args in upload_args:
+        with S3(s3root=s3root, inject_failure_rate=inject_failure_rate, upload_args=args) as s3:
+            s3urls = s3.put_many(objs)
+            assert list(dict(s3urls)) == list(dict(objs))
```

Comment:
What type is `s3urls` is there a reason we need to cast it to a `dict`?

Reply:
`s3.put_many` returns a list of tuples (which contain the keys and urls of the uploaded objects). I think the dictionary is being used to create a set? To avoid errors from mismatched url orders. Not super sure though